src/test-run/bookmark.ts

```diff
@@ -158,7 +158,13 @@ export default class TestRunBookmark {
 
             const preserveUrl = this.role.opts.preserveUrl;
 
-            await this._restorePage(this.role.redirectUrl as string, stateSnapshot);
+            if (this.role.redirectUrl) {
+                const redirectUrl = typeof this.role.redirectUrl === 'string'
```

Comment:
The `preserveUrl` is flag that manages the `redirectUrls` content. So, let's use it.
```js
const redirectUrl = this.opts.preserveUrl
    ? this.role.redirectUrl
    : this.role.redirectUrl[this.testRun.test.id]
```